src/internal/m365/collection/groups/collection.go

```diff
@@ -248,3 +249,211 @@ func (col *prefetchCollection[C, I]) streamItems(ctx context.Context, errs *faul
 
 	wg.Wait()
 }
+
+// -----------------------------------------------------------------------------
+// lazyFetchCollection
+// -----------------------------------------------------------------------------
+
+type lazyFetchCollection[C graph.GetIDer, I groupsItemer] struct {
+	data.BaseCollection
+	protectedResource string
+	stream            chan data.Item
+
+	contains container[C]
+
+	// added is a list of existing item IDs that were added to a container
+	added map[string]time.Time
+	// removed is a list of item IDs that were deleted from, or moved out, of a container
+	removed map[string]struct{}
+
+	getAndAugment getItemAndAugmentInfoer[C, I]
+
+	statusUpdater support.StatusUpdater
+}
+
+// Items utility function to asynchronously execute process to fill data channel with
```

Comment:
These comments are boilerplated from other ones that are a bit outdated and difficult to read.  Mind updating them?

Reply:
Ah yes. These are hard to read, lol. I'll fix them.